GUIDE.md

```diff
@@ -0,0 +1,198 @@
+will not apply to `y`.
+
+## Operations
+
+* **Addition** - both `Scalars` and `Matrices` support addition operation.
+    * For `Scalars` it is defined as $\hat{Q} \times \hat{Q} \to \hat{Q}$, and works as expected for rational numbers.
+    * For `Matrices` it is defined as $\hat{Q}^{N\times M} \times \hat{Q}^{N\times M} \to \hat{Q}^{N\times M}$ and works
+      element-wise.
+* **Subtraction** - both `Scalars` and `Matrices` support subtraction operation.
+    * For both `Scalars` and `Matrices` it is defined as $\hat{Q} \times \hat{Q} \to \hat{Q}$ and works like addition.
+* **Multiplication** - both `Scalars` and `Matrices` support multiplication operation.
+    * For `Scalars` it is defined as $\hat{Q} \times \hat{Q} \to \hat{Q}$, and works as expected for rational numbers.
+    * For `Matrices` it is defined as $\hat{Q}^{N\times M} \times \hat{Q}^{M\times K} \to \hat{Q}^{N\times K}$ and works
+      as expected for matrices.
+* **Division** - only `Scalars` support division operation.
+    * For `Scalars` it is defined as $\hat{Q} \times \hat{Q} \to \hat{Q}$, and works as expected for rational numbers.
+* **Inverse** - only `Matrices` support inverse operation.
+    * For `Matrices` it is defined as $\hat{Q}^{N\times N} \to \hat{Q}^{N\times N}$. Inverse $A^{-1}$ of matrix $A$ is
+      defined as $A^{-1}A = AA^{-1} = I$, where $I$ is identity matrix.
+* **Echelon** - only `Matrices` support echelon operation.
+    * For `Matrices` it is defined as $\hat{Q}^{N\times M} \to \hat{Q}^{N\times M}$. Echelon form is
+      defined [here](https://en.wikipedia.org/wiki/Row_echelon_form).
+* **Power** - both `Scalars` and `Matrices` support power operation.
+    * For `Scalars` it is defined as $\hat{Q} \times \hat{Q} \to \hat{Q}$, and works as expected for rational numbers.
+    * For `Matrices` it is defined as $\hat{Q}^{N\times N} \times \hat{Q} \to \hat{Q}^{N\times N}$. Power $A^k$ of
+      matrix $A$ is defined as $A^k = A \cdot A \cdot \dots \cdot A$ where $k$ is a positive integer.
+
+## Examples
+
+```matlab
+x = 1/3
+y = 2/3
+M = [1 2; 5 3]  % This syntax is not supported yet.
+N = [6 7; 3 1]  % Use GUI editor instead.
+% Addition
+z = x + y   % z = 1
+P = M + N   % P = [7 9; 8 4]
+% Subtraction
+z = x - y   % z = -1/3
+P = M - N   % P = [-5 -5; 2 2]
+% Multiplication
+z = x * y   % z = 2/9
+P = M * N   % P = [18 10; 36 22]
+% Division
+z = x / y   % z = 1/2
+% Inverse (only in GUI)
+P = inv(M)    % P = 1/7 * [-3 2; 5 -1]
+% Echelon (only in GUI)
+P = echelon(M)    % P = [1 0; 0 1]
+% Power
+z = x^2   % z = 1/9
+P = M^2   % P = [11 8; 20 19]
+```
+
+## Shell
+
+Shell is a command line interface for the calculator. It is used to input commands.
+Supported commands are:
+
+* `x = <expression>` - creates a variable `x` and assigns it the value of `<expression>`.
+* `<expression>` - evaluates `<expression>` and stores it in special variable `$`.
+  Error messages are displayed as a popup notification toast.
+
```

Comment:
It would be great to briefly describe expression syntax. I think pasting BNF rules from the code will be sufficient (Rust docs uses them too! see https://doc.rust-lang.org/reference/expressions.html).